### PR TITLE
Personal Access Token Feature

### PR DESCRIPTION
**Pull request by mohityadav766**
### Describe your changes :

- Added Apis for Personal Access Token
- Allow Developers to create Personal access token for use
#
### Type of change :

- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

==

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java [hunk 1]
```diff
@@ -1040,6 +1055,101 @@ public Response refreshToken(
...
+                  user.getEmail(),
+                  false,
+                  ServiceTokenType.PERSONAL_ACCESS,
+                  getExpiryDate(generateTokenRequest.getJWTTokenExpiry()));
+      PersonalAccessToken personalAccessToken = TokenUtil.getPersonalAccessToken(user, authMechanism);
+      tokenRepository.insertToken(personalAccessToken);
+
+      // Remove the previous tokens from DB and Cache
+      List<String> tokens = existingToken.stream().map((t) -> t.getToken().toString()).collect(Collectors.toList());
+      tokenRepository.deleteAllToken(tokens);
```
Comment (nahuelverdugo) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java:1146]: Could an empty list cause an error?

**Review by nahuelverdugo — APPROVED** [review]
I left some minor comments. I think we should test these kinds of flows where we play with the generation of tokens for users and are related to security.

File: openmetadata-spec/src/main/resources/json/schema/auth/emailVerificationToken.json [hunk 2]
```diff
@@ -9,12 +9,23 @@
...
       "type": "string",
       "enum": [
         "REFRESH_TOKEN",
         "EMAIL_VERIFICATION",
-        "PASSWORD_RESET"
+        "PASSWORD_RESET",
+        "PERSONAL_ACCESS"
+      ]
+    },
+    "serviceTokenType": {
```
[the remark this answers is not in the document]
Comment (mohityadav766) [reply openmetadata-spec/src/main/resources/json/schema/auth/emailVerificationToken.json:21]: no , those are for internal representation in user_tokens , Service tokens are the different type of token supported by OM -> OM_USER , BOT, PERSONAL_ACCESS

**Review by nahuelverdugo — DISMISSED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java [hunk 3]
```diff
@@ -3392,6 +3393,9 @@ public static TokenInterface getToken(TokenType type, String json) throws IOExce
         case REFRESH_TOKEN:
           resp = JsonUtils.readValue(json, RefreshToken.class);
           break;
+        case PERSONAL_ACCESS:
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java:3396]: lets call it as PERSONAL_ACCESS_TOKEN

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java [hunk 4]
```diff
@@ -1040,6 +1055,101 @@ public Response refreshToken(
     return Response.status(Response.Status.OK).entity(authHandler.getNewAccessToken(refreshRequest)).build();
   }
 
+  @GET
+  @Path("/token/personal")
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java:1059]: should we prefix /security/token under user, we can move our /token/

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java [hunk 5]
```diff
@@ -1040,6 +1055,101 @@ public Response refreshToken(
...
+        @ApiResponse(
+            responseCode = "200",
+            description = "The user ",
+            content =
+                @Content(mediaType = "application/json", schema = @Schema(implementation = PersonalAccessToken.class))),
+        @ApiResponse(responseCode = "400", description = "Bad request")
+      })
+  public Response revokePersonalAccessToken(@Context UriInfo uriInfo, @Context SecurityContext securityContext)
+      throws IOException {
+    String userName = securityContext.getUserPrincipal().getName();
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java:1099]: we should allow admin to revoke a user's token

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java (hunk 1 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java:1146]: nope

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java (hunk 3 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java]: done

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java (hunk 4 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java]: prefixed with /security

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java (hunk 5 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java]: updated, admin can view and revoke the users personal access token

**Review by nahuelverdugo — DISMISSED** [review]

**Review by harshach — APPROVED** [review]

**Pull request merged**